### PR TITLE
Add opt-in unsafe option transformation support

### PR DESCRIPTION
**Pull request by jchapuis**
This PR adds a `enableUnsafeOption` configuration setting, which allows for unsafe transformation from `Option[T1]` to `T2` if a transformer `T1 => T2` exists in scope.

This is really useful when working with proto3, where all fields are optional. See also issue #113.

If this can be considered for inclusion, I can work on adding coverage to the PR. We should also adapt the protobuf tests to proto3.

### CODE REVIEW

**Comment by krzemin** [comment]
Hi @jchapuis - your proposal looks good. Having this feature under optional flag seems as reasonable move.
Please continue working on coverage and docs - we can eventually merge this.

**Comment by jchapuis** [comment]
Great thanks, will continue working on it then.

**Comment by jchapuis** [comment]
Ok here you go, I've added coverage and documentation, let me know what you think!

**Comment by jchapuis** [comment]
Pinging you here again, am I on the right track, is there something else missing?

File: chimney/src/main/scala/io/scalaland/chimney/internal/TransformerMacros.scala [hunk 1]
```diff
@@ -108,7 +110,32 @@ trait TransformerMacros {
...
+      expandTransformerTree(q"$srcPrefixTree", config.rec)(From, toInnerT).mapRight { innerTransformer =>
+        q"_root_.scala.Option[$toInnerT]($innerTransformer)"
+      }
+    }
+  }
+
+  def expandSourceWrappedInOption(srcPrefixTree: Tree, config: Config)(From: Type,
+                                                                       To: Type): Either[Seq[DerivationError], Tree] = {
+    val fromInnerT = From.typeArgs.head
+    if (From <:< noneTpe) {
```
Comment (krzemin) [inline chimney/src/main/scala/io/scalaland/chimney/internal/TransformerMacros.scala]: That’s nice. Could you please add test case that checks this behavior?

File: chimney/src/main/scala/io/scalaland/chimney/internal/DerivationGuards.scala [hunk 2]
```diff
@@ -22,7 +22,11 @@ trait DerivationGuards {
   }
 
   def targetWrappedInOption(from: Type, to: Type): Boolean = {
-    to <:< optionTpe && to.typeArgs.head =:= from
+    to <:< optionTpe
+  }
+
+  def sourceWrappedInOption(from: Type, to: Type): Boolean = {
```
[the remark this answers is not in the document]
Comment (jchapuis) [reply chimney/src/main/scala/io/scalaland/chimney/internal/DerivationGuards.scala]: Yes absolutely. Good catch!

File: chimney/src/main/scala/io/scalaland/chimney/internal/TransformerMacros.scala (hunk 1 above)
Comment (jchapuis) [reply chimney/src/main/scala/io/scalaland/chimney/internal/TransformerMacros.scala]: Sure of course

**Comment by jchapuis** [comment]
Hey @krzemin I've worked on your two comments. Is there a way to compensate for the slight decrease in coverage? I think this occurs because there are more lines failing the compilation than before, right?

**Comment by jchapuis** [comment]
@krzemin thanks for merging! any estimate on when we can expect a release? Thanks! 🙏

**Comment by krzemin** [comment]
Tonight I preformed 0.3.3 release.

**Pull request merged**